solutions/platinum/cses-1742.mdx

```diff
@@ -0,0 +1,254 @@
+---
+id: cses-1742
+source: CSES
+title: Robot Path
+author: Mihnea Brebenel
+---
+
+## Explanation
+As you have already guessed this is a sweep line problem.
+
+Before we start talking about the actual solution, we need to make some observations about the problem's constraints. The coordinates can get easily big enough
+to cause us some issues. To solve this use [coordinate compresstion](/silver/sorting-custom?lang=cpp).
+
+But the main issue is how to find the first intersection when the robot stops ? To find the first intersection, we use binary search. The check function in binary search uses a Segment Tree as
+a sweep line. The beginning/ending of a line is marked as an event. The events are stored in the Segment Tree.
+
+The main issue of the problem is how to find the first intersection when the robot stops ? To find the first intersection we use binary search. The check function in binary search uses a Segment Tree as
+a sweep line. The beginning/ending of a line is marked as an event. The events are stored and processed in the Segment Tree.
+
+The last thing to do is to find the length of the incomplete segment - the on that causes the intersection. There are several ways to do this.
+The easiest way is to binary search its length. If you are not familiar with finding segment intersection, it may be worth solving [this](https://cses.fi/problemset/task/1740) first.
```

Comment:
```suggestion
The last thing to do is finding the length of the incomplete segment that causes the intersection. There are several ways to do this.

The easiest way is to binary search its length. If you are not familiar with finding segment intersections, it may be worth solving [this](https://cses.fi/problemset/task/1740) first.
```

mention the other ways too

Reply:
You should also link the USACO Guide editorial for that problem.

Reply:
Done.